### ISSUE TITLE
Build error on Debian

### ISSUE THREAD

**Issue by nsob1c12** [astra-toolbox/astra-toolbox#23]
Trying to build either 1.7.1beta or the latest git sources on Debian 8 (using either the system default Boost or v1.60.0 downloaded separately), the build stopped with the following error:

```
/usr/include/boost/assert.hpp:102:47: error: ‘noinline’ was not declared in this scope
           BOOST_NOINLINE void assertion_failed_msg(CharT const * expr, char const * msg, char const * function,
                                               ^
Makefile:290: recipe for target 'cuda/3d/mem3d.lo' failed
make: *** [cuda/3d/mem3d.lo] Error 1
```

I found that a workaround to enable the build to complete is to add `-DBOOST_NOINLINE='__attribute__ ((noinline))'` to the definition of `NVCCFLAGS` in the Makefile at line 41, but I do not know if this is likely to cause any other issues.

**Comment by wjp** [comment]
Which CUDA version are you using?

**Comment by nsob1c12** [comment]
I have version 6.0.37-5 of the nvidia-cuda-dev package installed, and running `nvcc --version` yields:

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2013 NVIDIA Corporation
Built on Thu_Mar_13_11:58:58_PDT_2014
Cuda compilation tools, release 6.0, V6.0.1
```

**Comment by wjp** [comment]
According to https://svn.boost.org/trac/boost/ticket/9392 , this is a problem that was fixed in CUDA 6.5. Your workaround sounds like it should be fine, though.

**Comment by nsob1c12** [comment]
Great, thanks.
